agent/utils/cpuinfo.go

```diff
@@ -0,0 +1,95 @@
+// +build linux
+
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package utils
+
+import (
+	"bufio"
+	"os"
+	"strings"
+)
+
+type CPUInfo struct {
+	Processors []Processor `json:"processors"`
+}
+
+type Processor struct {
+	Flags []string `json:"flags"`
+}
+
+var (
+	OpenFile = os.Open
+)
+
+func ReadCPUInfo(path string) (*CPUInfo, error) {
+	file, err := OpenFile(path)
+	if err != nil {
+		return nil, err
+	}
+
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+	var lines []string
+
+	for scanner.Scan() {
+		lines = append(lines, scanner.Text())
+	}
+
+	var cpuinfo = CPUInfo{}
+	var processor = &Processor{}
+
+	for i, line := range lines {
```

Comment:
You should be able to do this with only one scan loop without using the “lines” variable.

Reply:
Discussed offline. Scanner does not have an elegant way to meet my requirements. Will still keep lines variable here.